dm/dm/ctl/master/task_create.go

```diff
@@ -0,0 +1,138 @@
+// limitations under the License.
+
+package master
+
+import (
+	"bytes"
+	"context"
+	"net/http"
+	"os"
+
+	"github.com/pingcap/errors"
+	"github.com/pingcap/tiflow/dm/dm/config"
+	"github.com/pingcap/tiflow/dm/dm/ctl/common"
+	"github.com/pingcap/tiflow/dm/openapi"
+	"github.com/pingcap/tiflow/dm/pkg/log"
+	"github.com/pingcap/tiflow/dm/pkg/terror"
+	"github.com/spf13/cobra"
+	"go.uber.org/zap"
+)
+
+// NewTaskCreateCmd creates a Task Create command.
+func NewTaskCreateCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "create <config-file>",
+		Short: "Create a task by task configuration file",
+		RunE:  taskCreateFunc,
+	}
+	return cmd
+}
+
+func taskCreateFunc(cmd *cobra.Command, _ []string) error {
+	if len(cmd.Flags().Args()) != 1 {
+		cmd.SetOut(os.Stdout)
+		common.PrintCmdUsage(cmd)
+		return errors.New("please check output to see error")
+	}
+	res, err := createTask(cmd.Flags().Arg(0))
+	if err != nil {
+		return errors.Trace(err)
+	}
+	common.PrettyPrintOpenapiResp(true, res)
+	return nil
+}
+
+func createTask(filePath string) (*openapi.OperateTaskResponse, error) {
+	task, err := convertToOpenapiTaskFromFile(filePath)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	// create task
+	params := []interface{}{openapi.CreateTaskRequest{Task: *task}}
+	createResp := &openapi.OperateTaskResponse{}
+	sendError := common.SendOpenapiRequest(ctx, "DMAPICreateTask", params, http.StatusCreated, createResp)
+	if sendError != nil {
+		return nil, errors.Trace(sendError)
+	}
+	return createResp, nil
+}
+
+func convertToOpenapiTaskFromFile(filePath string) (*openapi.Task, error) {
+	var (
+		err         error
+		sendError   *common.SendToOpenapiError
+		fileContent []byte
+		taskStr     string
+		params      []interface{}
+	)
+	fileContent, err = common.GetFileContent(filePath)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	task := config.NewTaskConfig()
+	err = task.RawDecode(string(fileContent))
+	if err != nil {
+		return nil, err
+	}
+	if task.TargetDB != nil && task.TargetDB.Security != nil {
+		loadErr := task.TargetDB.Security.LoadTLSContent()
+		if loadErr != nil {
+			log.L().Warn("load tls content failed", zap.Error(terror.ErrCtlLoadTLSCfg.Generate(loadErr)))
+		}
+		fileContent = []byte(task.String())
+	}
+
+	lines := bytes.Split(fileContent, []byte("\n"))
+	// we check if `is-sharding` is explicitly set, to distinguish between `false` from default value
+	isShardingSet := false
+	for i := range lines {
+		if bytes.HasPrefix(lines[i], []byte("is-sharding")) {
+			isShardingSet = true
+			break
+		}
+	}
+	// we check if `shard-mode` is explicitly set, to distinguish between "" from default value
+	shardModeSet := false
+	for i := range lines {
+		if bytes.HasPrefix(lines[i], []byte("shard-mode")) {
+			shardModeSet = true
+			break
+		}
+	}
+
+	if isShardingSet && !task.IsSharding && task.ShardMode != "" {
```

Comment:
Did you mean:
```
is-sharding: false
shard-mode = "xxx"
```
If `is-sharding` is deprecated, what about setting the `is-sharding` to `true`? Is it legal?

Reply:
this is the same as `start-task` at [here](https://github.com/pingcap/tiflow/blob/master/dm/dm/ctl/master/start_task.go#L90) to make `task create/start` compatible with `start-task` behavior. 
bwt, taking this out of `start-task` is because we may just maintain commands which call openapi interface and we may extend the commands behavior but grpc interface does not support in the future, but at present, it is better to be compatible.

Reply:
i find out this [pr](https://github.com/pingcap/tiflow/pull/5223), maybe it is the aim which will return waring or error.